Indexer.php

```diff
@@ -73,13 +73,91 @@ public function isUrlIndexedandNotExpired($uri, array $metadata = [])
 
         $query = $this->client->createSelect();
         $query->setQuery($queryPhrase);
-
         $result = $this->client->select($query);
 
         return ($result->getNumFound() > 0);
     }
 
     /**
+     * Returns unique hashes for uris from one page.
+     *
+     * @param $uris
+     *
+     * @return array
+     */
+    public function getUniqueHashIds($uris)
```

Comment:
That function can be refactored. You add $HashId in array key end later you retrieving keys from arrays. Why you just do not use syntax like $hashes[] = $hashId; return $hashes;